api/bsf_handler.go

```diff
@@ -12,26 +12,27 @@ import (
 	"github.com/web-platform-tests/wpt.fyi/shared"
 )
 
+// BSFHandler is an http.Handler for the /api/bsf endpoint.
+type BSFHandler struct {
+	fetcher shared.FetchBSF
+}
+
 // apiBSFHandler fetches browser-specific failure data based on the URL params.
 func apiBSFHandler(w http.ResponseWriter, r *http.Request) {
-	handleBSF(w, r, shared.NewFetchBSF())
+	ctx := r.Context()
+	// Serve cached with 5 minute expiry. Delegate to BSFHandler on cache miss.
```

Comment:
Given the frequency of the cron job (weekly), I think it's safe to set the expiry to 1 hour.